### PR TITLE
dird: ignore duplicate job checking on virtual fulls started by consolidation

### PR DESCRIPTION
**Pull request by fbergkemper**
- Ignore AllowDuplicateJobs directive on generated virtual backup jobs by a consolidate job
- The test checks if we end up with a canceled virtual full after consolidation. The test fails if the virtual full is canceled.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [x] Decision taken that a system- or unittest is required (if not, then remove this paragraph)
- [x] The decision towards a systemtest is reasonable compared to a unittest
- [x] Testname matches exactly what is being tested
- [x] Output of the test leads quickly to the origin of the fault

### CODE REVIEW

**Comment by fbergkemper** [comment]
The patch has side effects and is not a solution. The implementation of a real solution is being put on hold for the time being but will follow in near future.

File: systemtests/CMakeLists.txt [hunk 1]
```diff
@@ -623,6 +623,7 @@ set(SYSTEM_TESTS
     virtualfull
     virtualfull-bscan
     volume-pruning
+    ai-duplicate-job-cancel
```
Comment (pstorz) [inline systemtests/CMakeLists.txt:626]: If not already done, we should order the list of tests alphabetically.
Also, the testname does not match what is being tested.
How about 
**ai-consolidation-ignores-duplicate-job-checking**

**Review by pstorz — CHANGES_REQUESTED** [review]
Good work!

The testname does not exactly say what it verifies, so that should be adapted.

File: systemtests/CMakeLists.txt (hunk 1 above)
Comment (fbergkemper) [reply systemtests/CMakeLists.txt]: Testname changed to "ai-consolidate-ignore-duplicate-job". Also sorted the whole list alphabetically.

**Comment by fbergkemper** [comment]
Made some notes in the documentation on how the Allow Duplicate Jobs directive behaves in case of generated virtual backups by a consolidate job. See additional commit https://github.com/bareos/bareos/pull/552/commits/1eb3a9a1ca2e8322fe9ea736024b4c82033ae5f3.

File: docs/manuals/source/TasksAndConcepts/AlwaysIncrementalBackupScheme.rst [hunk 2]
```diff
@@ -137,6 +137,7 @@ The :config:option:`dir/job = Consolidate`\  job evaluates all jobs configured w
 
 The always incremental jobs need to be executed during the backup window (usually at night), while the consolidation jobs should be scheduled during the daytime when no backups are executed.
 
+In principle generated virtual backup jobs of a consolidation job are not affected by :config:option:`dir/job/AllowDuplicateJobs`\ settings.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/AlwaysIncrementalBackupScheme.rst:140]:
```suggestion
Virtual backup jobs of a consolidation are not affected by :config:option:`dir/job/AllowDuplicateJobs`\ settings.
```

File: docs/manuals/source/manually_added_config_directive_descriptions/dir-job-AllowDuplicateJobs.rst.inc [hunk 3]
```diff
@@ -24,3 +24,5 @@ If this directive is enabled duplicate jobs will be run. If the directive is set
    :config:option:`dir/job/CancelRunningDuplicates`\
 
 If none of these directives is set to :strong:`yes`, Allow Duplicate Jobs is set to :strong:`no` and two jobs are present, then the current job (the second one started) will be cancelled.
+
+In principle generated virtual backup jobs of a consolidation job are not affected by the directive. In those cases the directive is going to be ignored.
```
Comment (pstorz) [inline docs/manuals/source/manually_added_config_directive_descriptions/dir-job-AllowDuplicateJobs.rst.inc:28]:
```suggestion
Virtual backup jobs of a consolidation are not affected by the directive. In those cases the directive is going to be ignored.
```

**Review by pstorz — CHANGES_REQUESTED** [review]

File: docs/manuals/source/TasksAndConcepts/AlwaysIncrementalBackupScheme.rst (hunk 2 above)
Comment (fbergkemper) [reply docs/manuals/source/TasksAndConcepts/AlwaysIncrementalBackupScheme.rst]: changed

File: docs/manuals/source/manually_added_config_directive_descriptions/dir-job-AllowDuplicateJobs.rst.inc (hunk 3 above)
Comment (fbergkemper) [reply docs/manuals/source/manually_added_config_directive_descriptions/dir-job-AllowDuplicateJobs.rst.inc]: changed

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**